File: glitch/backend/app/schema/activity.py
from pydantic import BaseModel
from typing import Optional


class Activity(BaseModel):
    rid: int
    activity: str
    datetime_entry: str
    datetime_update: str
    rid_users: int
    name: Optional[str] = None

    class Config:
        orm_mode = True


class ActivityCreate(BaseModel):
    rid_items: int
    rid_users: int
    activity: str

    class Config:
        orm_mode = True


class ActivityUpdate(BaseModel):
    rid: int
    activity: str

    class Config:
        orm_mode = True

File: glitch/frontend/app/glitch/src/types/Activity.ts
export interface Activity {
  rid: number
  activity: string
  datetime_entry: string
  datetime_update: string
  rid_users: number
  name: string
}

export interface ActivityCreate {
  rid_items: number
  rid_users: number
  activity: string
}

export interface ActivityUpdate {
  rid: number
  activity: string
}
